bindings/python/py_src/tokenizers/decoders/__init__.pyi

```diff
@@ -1,65 +1,87 @@
-from typing import List
-
+# Generated content DO NOT EDIT
 class Decoder:
-    """Base class for all decoders
+    """
+    Base class for all decoders
 
     This class is not supposed to be instantiated directly. Instead, any implementation of
     a Decoder will return an instance of this class when instantiated.
     """
 
-    def decode(self, tokens: List[str]) -> str:
-        """ Decode the given list of string to a final string """
+    def decode(self, tokens):
+        """
+        Decode the given list of string to a final string
+        """
         pass
 
-class ByteLevel(Decoder):
-    """ ByteLevel Decoder """
+class BPEDecoder(Decoder):
+    """
+    Instantiate a new BPEDecoder
 
-    def __init__(self) -> None:
-        """ Instantiate a new ByteLevel Decoder """
-        pass
+    Args:
+        suffix: str:
+            The suffix that was used to caracterize an end-of-word. This suffix will
+            be replaced by whitespaces during the decoding
+    """
 
-class WordPiece(Decoder):
-    """ WordPiece Decoder """
+    def __init__(self, suffix="</w>"):
+        pass
+    def decode(self, tokens):
```

Comment:
All the inherited methods are duplicated on the children classes. I think we should remove them as the various IDE should be able to understand this by themselves. (Just like they would with plain Python files)

Reply:
That makes a lot more checks (checking mro, uptree for existing signatures, checking signatures are the same and so one).I I really don't think it's worth it as it's generated code now. The possibility of a bug that shadows existing functions outweighs the benefit of writing less generated code in my mind.